gymnasium/envs/mujoco/mujoco_rendering.py

```diff
@@ -281,6 +281,10 @@ def render(
                         seg_ids[geom.segid + 1, 1] = geom.objid
                 rgb_img = seg_ids[seg_img]
 
+            if render_mode == "rgbd_array":
```

Comment:
This section (L259-L289) was not written to support 3 render types, it should be re-written to be clearer, into 2 stages, first stage collects the images, and the second stage returns the correct images.
```py
if render_mode in ["depth_array", "rgbd_array"]:
    depth_img = depth_arr.reshape(self.viewport.height, self.viewport.width)
if render_mode in ["rgb_array", "rgbd_array"]:
    ...
```

```py
if render_mode == "rgb_array":
    return ...
if render_mode == "depth_array":
    return ...
if render_mode == "rgbd_array":
    return ...
```